src/node/setupServer.ts

```diff
@@ -9,12 +9,23 @@ import { getResponse } from '../utils/getResponse'
 import { parseRequestBody } from '../utils/parseRequestBody'
 import * as requestHandlerUtils from '../utils/requestHandlerUtils'
 
+const isNode =
+  Object.prototype.toString.call(
+    typeof process !== 'undefined' ? process : 0,
+  ) === '[object process]'
+
 /**
  * Sets up a server-side requests interception with the given mock definition.
  */
 export const setupServer = (...requestHandlers: RequestHandlersList) => {
   const interceptor = new RequestInterceptor()
 
+  if (!isNode) {
+    console.error(
+      '[MSW] It appears you are trying to set up Node Integration in the browser. Please make sure to move any calls to setupServer so they only run in a node/jsdom based testing environment',
+    )
+    throw 'Cannot call setupServer in the browser'
```

Comment:
Do you think we should throw in this case?
This will be an unhandled exception on the developer's side, very likely crashing their app. I think that we shouldn't throw at this case, but rather:

1. Produce an error (like you're doing with `console.error`.
1. Have an explicit `return` within this `if` statement. This way calling `setupServer` in a non-Node environment won't return its usual API (`.listen`, `.use`, etc.), and will produce an error instead.

Reply:
My thinking was throw to fail fast and crash rather than weird unexpected behavior, but I was back and forth on that myself. It's not a strongly held opinion, so I'll change it to remove the throw if that's what you prefer.

Reply:
It should be fine to produce an error + don’t execute any code intended for Node (due to explicit return).

I’m thinking if we should add a similar check and error when running `setupWorker` in Node?